docs/graphx-programming-guide.md

```diff
@@ -708,7 +708,9 @@ messages remaining.
 > messaging function.  These constraints allow additional optimization within GraphX.
 
 The following is the type signature of the [Pregel operator][GraphOps.pregel] as well as a *sketch*
-of its implementation (note calls to graph.cache have been removed):
+of its implementation (note: to avoid stackOverflowError due to long lineage chains, graph and 
```

Comment:
I think @viirya is suggesting we remove all references to checkpointing from this implementation sketch. @viirya, correct me if I'm wrong. I suggest reverting these changes to `graphx-programming-guide.md`. Just add a brief note that GraphX periodically checkpoints the graph and message lineages, and this checkpoint interval can be configured with the `spark.graphx.pregel.checkpointInterval` Spark configuration property.

@viirya @felixcheung Should we document this config property in the Spark configuration document as well?